src/subscription_manager/repofile.py

```diff
@@ -151,9 +154,13 @@ def from_ent_cert_content(cls, content, baseurl, ca_cert, release_source):
         repo['sslclientcert'] = content.cert.path
         repo['sslcacert'] = ca_cert
         repo['metadata_expire'] = content.metadata_expire
+        if 'arches' in repo and len(content.arches) > 0:
+            if isinstance(content.arches[0], "".__class__):
+                repo['arches'] = content.arches
+            else:
+                repo['arches'] = [x.encode('UTF-8') for x in content.arches]
```

Comment:
this seems to imply that `repo['arches']` can be both a list of arches and a space-separated string of arches, and later on `fix_content()` actually changes the type (see my comment about it); I'd avoid changing the type of a key, and always have it either as `None` or as list of arches

Reply:
This line is important for Py2/Py3 support.
In case `content.arches[x]` is already an instance of the corresponding string class type (unicode or bytes depending on the Python version), `repo['arches']` just takes its value.
If the string class type does not fit, it is converted accordingly.

Reply:
note that sub-man is py3-only, so py2 code is not needed; also, I'd keep `repo['arches']` as list only, instead of switching from list to string

Reply:
What do you mean by switching to string? `[x.encode('UTF-8') for x in content.arches]` produces a list, but changes from bytes &rarr; unicode and vice versa.